aws-ssm-document/src/main/java/com/amazonaws/ssm/document/DocumentExceptionTranslator.java

```diff
@@ -63,7 +64,7 @@ RuntimeException getCfnException(@NonNull final SsmException e, @NonNull String
 
             return new CfnInvalidRequestException(e.getMessage(), e);
 
-        } else if (e instanceof InvalidDocumentException) {
+        } else if (e instanceof InvalidDocumentException || e instanceof InvalidResourceIdException) {
```

Comment:
will there be an edge case when invalid resource id does not mean not found?

Reply:
Not at this time. This is the error thrown by our validation of tagging requests
There should be 2 scenarios that could throw this exception:
1) The resourceId failed validation against the model, which means the id is not viable
2) The resource could not be found and this is the error thrown
In either case the resource does not exist